orttraining/orttraining/test/python/orttraining_test_orttrainer_bert_toy_onnx.py

```diff
@@ -1,75 +1,515 @@
+        },
+    })
+   
+    seed = 1
+    torch.manual_seed(seed)
+    set_seed(seed)
+    trainer = orttrainer.ORTTrainer(model, model_desc, optim_config, options=opts)
+    
+    experimental_losses = []
+    for i in range(train_steps):
+        sample_input = generate_random_input_from_model_desc(model_desc, i)
+
+        experimental_losses.append(trainer.train_step(*sample_input).cpu().item())
+
+    # LEGACY IMPLEMENTATION
+    device = torch.device("cuda", 0)
+    legacy_model_desc, learning_rate_description, learning_rate = legacy_model_params() 
     seed = 1
     torch.manual_seed(seed)
     set_seed(seed)
+    
+    legacy_trainer = Legacy_ORTTrainer(model, None, legacy_model_desc, "LambOptimizer",
+                       None,
+                       learning_rate_description,
+                       device)
+    legacy_losses = []
+    for i in range(train_steps):
+        sample_input = generate_random_input_from_model_desc(model_desc, i)
+        legacy_sample_input = [*sample_input, learning_rate]
 
-    # Modeling
-    pytorch_transformer_path = os.path.join('..', '..', '..', 'onnxruntime', 'test', 'testdata')
-    bert_onnx_model_path = os.path.join(pytorch_transformer_path, "bert_toy_postprocessed.onnx")
-    model = onnx.load(bert_onnx_model_path)
+        legacy_losses.append(legacy_trainer.train_step(legacy_sample_input).cpu().item())
+
+    _test_helpers.assert_model_outputs(experimental_losses, legacy_losses, True, rtol=1e-4)
+
+@pytest.mark.parametrize("initial_lr, lr_scheduler, legacy_lr_scheduler", [
+    (1.0, optim.lr_scheduler.ConstantWarmupLRScheduler, legacy_constant_lr_scheduler_1),
+    (0.5, optim.lr_scheduler.ConstantWarmupLRScheduler, legacy_constant_lr_scheduler_5),
+    (1.0, optim.lr_scheduler.CosineWarmupLRScheduler, legacy_cosine_lr_scheduler),
+    (1.0, optim.lr_scheduler.LinearWarmupLRScheduler, legacy_linear_lr_scheduler),
+    (1.0, optim.lr_scheduler.PolyWarmupLRScheduler, legacy_poly_lr_scheduler),
+])
+def testToyBERTModelLegacyExperimentalLRScheduler(initial_lr, lr_scheduler, legacy_lr_scheduler):
+    # EXPERIMENTAL API
     model_desc = bert_model_description()
+    model = load_bert_onnx_model()
+
+    total_steps = 10
+    optim_config = optim.LambConfig(lr=initial_lr)
+    opts =  orttrainer.ORTTrainerOptions({
+        'debug' : {
+            'deterministic_compute': True
+        },
+        'lr_scheduler' : lr_scheduler(total_steps=total_steps, warmup=0.5)
+    })
+    
+    torch.manual_seed(1)
+    set_seed(1)
+
+    trainer = orttrainer.ORTTrainer(model, model_desc, optim_config, options=opts)
+  
+    experimental_losses = []
+    for i in range(total_steps):
+        sample_input = generate_random_input_from_model_desc(model_desc, i)
+        experimental_losses.append(trainer.train_step(*sample_input).cpu().item())
+        
+        assert trainer.options.lr_scheduler.get_last_lr()[0] == legacy_lr_scheduler(i)
+
+    # LEGACY IMPLEMENTATION
+    device = torch.device("cuda", 0)
```

Comment:
Reuse the device idea from previous comment in this test too. Relying on default values can be misleading